02-microarray/differential-expression_microarray_02_several-groups.Rmd

```diff
@@ -0,0 +1,456 @@
+  BiocManager::install("EnhancedVolcano", update = FALSE)
+}
+if (!("apeglm" %in% installed.packages())) {
+  # Install this package if it isn't installed yet
+  BiocManager::install("apeglm", update = FALSE)
+}
+```
+
+Attach the packages we need for this analysis.
+
+```{r}
+# Attach the library
+library(limma)
+
+# We will need this so we can use the pipe: %>%
+library(magrittr)
+
+# We'll use this for plotting
+library(ggplot2)
+```
+
+## Import and set up data
+
+Data downloaded from refine.bio include a metadata tab separated values (TSV) file and a data TSV file. 
+This chunk of code will read the both TSV files and add them as data frames to your environment.
+
+```{r}
+# Read in metadata TSV file
+metadata <- readr::read_tsv(file.path(
+  data_dir, # Replace with path to your metadata file
+  "metadata_GSE37418.tsv" # Replace with the name of your metadata file
+))
+
+# Read in data TSV file
+df <- readr::read_tsv(file.path(
+  data_dir, # Replace with path to your data file
+  "GSE37418.tsv" # Replace with the name of your data file
+)) %>%
+  # Tuck away the Gene id column as rownames
+  tibble::column_to_rownames("Gene")
+```
+
+## Removing groups that are too small
+
+We will be using the `subgroup` variable labels in our metadata to test differentially expression across.
+Let's take a look at how many samples of each subgroup we have. 
+
+```{r}
+summary(as.factor(metadata$subgroup)) # Notice we are coercing this to a factor so we can see the counts per group, `subgroup` is actually a character
+```
+
+Looks like there is one sample that has been labeled by the authors as an outlier (`SHH OUTLIIER`), as well as one group, `U`, that only has two samples. 
+We will probably want to remove the `U` samples and this outlier since their inclusion might throw off our differential expression analysis results. 
+
+Let's start out by removing the outlier and the `U` group, we can do this all at once by removing groups smaller than 3. 
+
+```{r}
+filtered_metadata <- metadata %>%
+  dplyr::group_by(subgroup) %>%
+  dplyr::filter(dplyr::n() > 3) %>%
+  dplyr::ungroup()
+```
+
+Let's take a look at the subgroup summary again. 
+
+```{r}
+summary(as.factor(filtered_metadata$subgroup))
+```
+
+Note that the `U` and the `SHH OUTLIER` samples are gone and only the four groups we are interested in are left. 
+
+But, we still need to filter these samples out from the expression data that's stored in `df`. 
+
+```{r}
+# Make the data in the order of the metadata
+df <- df %>%
+  dplyr::select(filtered_metadata$geo_accession)
+
+# Check if this is in the same order
+all.equal(colnames(df), filtered_metadata$geo_accession)
+```
+
+## Create the design matrix
+
+`limma` needs a numeric design matrix to signify which samples are of which subtype of medulloblastoma. 
+Now we will create a model matrix based on our `subgroup` variable.
+We are using a `+ 0` in the model so the first group is not treated as your intercept. 
```

Comment:
I'm not sure what the exact right explanation is here. I've been looking at different sources online and my understanding of what intercepts represent is a little bit in conflict with what the limma user's guide shows. But I'm just trusting the limma user's guide. But I can't just say that.